src/commands/Command.ts

```diff
@@ -0,0 +1,46 @@
+import { SlashCommandBuilder, ChatInputCommandInteraction, Message } from "discord.js";
+
+export type SlashCommandHandler = (interaction: ChatInputCommandInteraction) => any;
+export type SlashCommandResult = () => Omit<SlashCommandBuilder, "addSubcommand" | "addSubcommandGroup">;
+
+export interface CommandConfiguration {
+    name: string;
+    emojiName?: string;
```

Comment:
If `emojiName` is referenced in `ReactionCommand.ts`, which extends this interface, do we need it here as well?

Reply:
Similar question regarding `execute` and `executeSlashCommand` - we might need to prune the higher level interface a bit

Reply:
It looks like the emojiName variable here was an artefact from the restructure; it wasn't actually used in the Command class. Removed.

execute and executeSlashCommand are definitely redundant; I wasn't sure if there was a reason for the redundancy while migrating so I kept both until I was sure. I will subclass the Command class for slash commands - this would have made more sense for me to do from the start as there's already subclasses for the other command types.